src/client.ts

```diff
@@ -178,73 +214,129 @@ export class Client {
-          dbStatementSerializer: RedisDbStatementSerializer
-        }),
-        new IORedisInstrumentation({
-          dbStatementSerializer: RedisDbStatementSerializer
-        }),
-        new PrismaInstrumentation({
-          middleware: true
-        })
-      ]
+        }
+      },
+      "@opentelemetry/instrumentation-http": {
+        headersToSpanAttributes: {
+          server: { requestHeaders }
+        }
+      },
+      "@opentelemetry/instrumentation-ioredis": {
+        dbStatementSerializer: RedisDbStatementSerializer
+      },
+      "@opentelemetry/instrumentation-koa": {
+        requestHook: function (span: OtelSpan, info) {
+          if (sendParams) {
+            // Request parameters to magic attributes
+            const queryParams = info.context.request.query
+
+            span.setAttribute(
+              "appsignal.request.parameters",
+              JSON.stringify(queryParams)
+            )
+          }
+        }
+      },
+      "@opentelemetry/instrumentation-redis": {
+        dbStatementSerializer: RedisDbStatementSerializer
+      },
+      "@opentelemetry/instrumentation-redis-4": {
+        dbStatementSerializer: RedisDbStatementSerializer
+      },
+      "@prisma/instrumentation": {
+        middleware: true
+      }
+    }
+  }
+
+  private instrumentations(
+    instrumentationsOption: AppsignalInstrumentationOption
+  ): InstrumentationOption[] {
+    const defaultConfig = this.defaultInstrumentationsConfig()
+
+    const defaultInstrumentations = Object.entries(DefaultInstrumentationMap)
+      .filter(([key, _instrumentationConstructor]) => {
+        const instrumentationName =
+          key as keyof typeof DefaultInstrumentationMap
+
+        const userInstrumentationConfig =
+          instrumentationsOption[instrumentationName] || {}
+
+        if (userInstrumentationConfig instanceof InstrumentationBase) {
+          return false
+        }
+
+        return userInstrumentationConfig.enabled !== false
+      })
+      .map(([key, instrumentationConstructor]) => {
+        const instrumentationName =
+          key as keyof typeof DefaultInstrumentationMap
+
+        const userInstrumentationConfig =
+          instrumentationsOption[instrumentationName] || {}
+        const defaultInstrumentationConfig =
+          defaultConfig[instrumentationName] || {}
+
+        const instrumentationConfig = {
+          ...defaultInstrumentationConfig,
+          ...userInstrumentationConfig
+        }
+
+        return new instrumentationConstructor(instrumentationConfig)
+      })
+
+    const defaultInstrumentationNames = Object.keys(DefaultInstrumentationMap)
+
+    const additionalInstrumentations = Object.entries(instrumentationsOption)
+      .filter(([key, instrumentation]) => {
+        const isDefault = defaultInstrumentationNames.includes(key)
+        const isInstance = instrumentation instanceof InstrumentationBase
+
+        return !isDefault || isInstance
```

Comment:
This allows for default instrumentation that are configured with an instance to be added as part of the additionalInstrumentations, right? Would they be added twice in that case?

Reply:
They shouldn't be added twice, because they're filtered out from `defaultInstrumentations` above if they're instances.